src/content/1.7/modules/sample_modules/view-order-hooks-demo.md

```diff
@@ -0,0 +1,721 @@
+        $this->author = 'PrestaShop';
+        $this->version = '1.0.0';
+        $this->ps_versions_compliancy = ['min' => '1.7.7.0', 'max' => _PS_VERSION_];
+
+        parent::__construct();
+
+        $this->displayName = $this->l('Demo view order hooks');
+        $this->description = $this->l('Demonstration of new hooks in PrestaShop 1.7.7 order view page');
+    }
+
+    public function install()
+    {
+        if (!parent::install()) {
+            return false;
+        }
+
+        $installer = InstallerFactory::create();
+
+        return $installer->install($this);
+    }
+
+    public function uninstall()
+    {
+        $installer = InstallerFactory::create();
+
+        return $installer->uninstall() && parent::uninstall();
+    }
+
+}
+```
+
+Let's create `Signature` entity class and use Doctrine Object Relational Mapping (ORM) annotations.
+For more information: https://devdocs.prestashop.com/1.7/modules/concepts/doctrine/#define-an-entity
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table()
+ * @ORM\Entity()
+ */
+class Signature
+{
+    /**
+     * @var int|null
+     *
+     * @ORM\Id
+     * @ORM\Column(name="id_signature", type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(name="id_order", type="integer")
+     */
+    private $orderId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(name="filename", type="string")
+     */
+    private $filename;
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Signature
+     */
+    public function setId(int $id): Signature
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @return Signature
+     */
+    public function setFilename(string $filename): Signature
+    {
+        $this->filename = $filename;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+}
+
+```
+
+Lets create `SignatureRepository` class:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Repository;
+
+use Doctrine\ORM\EntityRepository;
+
+class SignatureRepository extends EntityRepository
+{
+}
+```
+
+Let's put our signature picture, `john_doe.png` inside `/signatures/` folder.
+
+ {{< figure src="../img/view-order-hooks-demo/john_doe.png" title="Signature" >}}
+
+Let's create Order Repository and data structures for interacting with Orders data:
+
+Data Transfer Object `Order.php` in `src/DTO`
```

Comment:
Please explain briefly what is a DTO (you can use external reference for that) and why it's a good thing (in comparison to arrays)